assessment/libs/stats.php

```diff
@@ -2401,4 +2402,420 @@ function dotplot($a,$label,$dotspace=1,$labelspace=null,$width=300,$height=150)
 	return showasciisvg($outst,$width,$height);
   }
 
+//---------------------------------------------ANOVA-Oneway F ratio-----------------------------------------
+// Function: anova1way_f(arr1,arr2, [arr3,...])
+// Returns F ratio and the corresponding P value as an array. 
+//
+// Parameters:
+// arr1, arr2, ...: Arrays in the form [2,3,4,5,...]; it also accepts unequal sample sizes. 
+//  
+// Returns:
+// F ratio and the corresponding P value as an array in the form [F ratio, P value].
+
+function anova1way_f(... $arr){
+	
+	$n=array();  
+	foreach($arr as $a){
+		if (!is_array($a)) { $a = explode(',',$a);};
+		$n[]=count($a);
+		/*if (count(array_unique($n))!=1) {
+			echo "Error: ANOVA requires the same length for all arrays";
+			return false;
+		}*/
+	}
+	if (count($n)<3) {
+		echo "Error: ANOVA requires three or more arrays";
+		return false;
+	}
+	$N=array_sum($n);	
+	$numargs = func_num_args();
+	//$args=func_get_args();
+	$mean=array();
+	$ss=array();
+	for($i=0;$i<$numargs;$i++){
+		$mean[$i]=mean($arr[$i]);
+		$ss[$i]=variance($arr[$i])*(count($arr[$i])-1);
+		//$n[$i]=count($args[$i]);
+	}
+	for ($i=0;$i<$numargs;$i++){
+		$gmean=array_sum($mean)/count($mean); //grand mean 

```

Comment:
This loop doesn't make sense to me.  It looks like you're calculating exactly the same thing multiple times.